x/datadeal/types/deal.go

```diff
@@ -51,9 +54,19 @@ func (m Deal) ValidateBasic() error {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "budget is not a valid Coin object")
 	}
 
+	if m.CurNumData <= 0 {
```

Comment:
My guess is that `curNumData` can be zero.
Therefore, it seems to have to be modified to be less than 0.
```suggestion
	if m.CurNumData < 0 {
```

Reply:
Lint says we don't need to do that check in case of `uint64`. So, I delete that check code.